### PR TITLE
Add docs indicating that the onnxruntime engine from other distributions can be compatible with the WinRT NuGet

### PR DESCRIPTION
**Pull request by smk2007**
Add docs indicating that the onnxruntime engine from other distributions can be compatible with the WinRT NuGet

### CODE REVIEW

File: csharp/src/Microsoft.AI.MachineLearning/Microsoft.AI.MachineLearning.Rules.Project.xml [hunk 1]
```diff
@@ -8,13 +8,13 @@
     <DataSource Persistence="ProjectFile" HasConfigurationCondition="false" Label="Globals" />
   </Rule.DataSource>
 
-  <StringProperty Name="OnnxRuntimePathOverride"
-                  DisplayName="Custom ONNXRuntime DLL"
-                  Description="Specifies a path to a custom OnnxRuntime dll."
-                  Category="General" />
+  <BoolProperty Name="DisableOnnxRuntimeDllCopy"
+                DisplayName="Disable onnxruntime.dll copy on build"
+                Description="Suppresses binplacing the onnxruntime.dll from the Mirosoft.AI.MachineLearning NuGet distribution."
```
Comment (ryanlai2) [inline csharp/src/Microsoft.AI.MachineLearning/Microsoft.AI.MachineLearning.Rules.Project.xml:13]: >onnxruntime [](start = 55, length = 11)

In what scenario would onnxruntime.dll not be binplaced? Doesn't Microsoft.AI.MachineLearning.dll depend on onnxruntime.dll?

Comment (smk2007) [reply csharp/src/Microsoft.AI.MachineLearning/Microsoft.AI.MachineLearning.Rules.Project.xml:13]: When the app developer wants to binplace another one.

**Review by ryanlai2 — DISMISSED** [review]
:shipit:

File: docs/InferenceHighLevelDesign.md [hunk 2]
```diff
@@ -109,10 +109,27 @@ Developers can now choose which API works best for their scenario.
...
 |Compatability testing| Tested with OS flights against supported GPU's and CPU's | App performs compatibility testing|
 |Opset| Refreshed in OS updates| App chooses|
+
+ 
+### Using the NuGet WinRT API with other C-API distributions
+The WinRT API NuGet is distributed with a curated build of the OnnxRuntime engine. App developers may wish to use the WinRT API, but find themselves limited to the functionality provided by the curated OnnxRuntime engine distributed as part of the WinRT API NuGet package. This can happen because the OnnxRuntime engine shipped with the WinRT API NuGet package only contains the CPU and DML execution providers.
+
+App developers may additionally wish to use a custom build-from-source version of the OnnxRuntime engine as well, or use a prebuilt version of the OnnxRuntime engine from another distribution source like the Micorosoft.ML.OnnxRuntime.MKLML distribution.
+
+To enable this, the WinRT API NuGet has been made to be compatible with a set of OnnxRuntime engines that ship in different Nuget packages.
```
Comment (martinb35) [inline docs/InferenceHighLevelDesign.md:126]: typo "NuGet"

**Review by martinb35 — COMMENTED** [review]
We should get Nick and Paul to look at this too.

File: docs/InferenceHighLevelDesign.md [hunk 3]
```diff
@@ -109,10 +109,27 @@ Developers can now choose which API works best for their scenario.
...
+
+Please refer to the following table listing the distributions with compatible OnnxRuntime engines.
+- [Microsoft.ML.OnnxRuntime](https://www.nuget.org/packages/Microsoft.ML.OnnxRuntime)
+- [Microsoft.ML.OnnxRuntime.DirectML](https://www.nuget.org/packages/Microsoft.ML.OnnxRuntime.DirectML/)
+- [Microsoft.ML.OnnxRuntime.MKLML](https://www.nuget.org/packages/Microsoft.ML.OnnxRuntime.MKLML)
+
+Note that compatible distributions must match in release version.
+
+In order to use compatilbe engines, replace the onnxruntime.dll with the desired engine binary and its associated binaries.
```
Comment (smk2007) [inline docs/InferenceHighLevelDesign.md:135]: compatible

**Review by ngeisler11 — APPROVED** [review]

**Review by ngeisler11 — APPROVED** [review]

**Review by ryanlai2 — APPROVED** [review]

**Review by martinb35 — APPROVED** [review]

**Pull request merged**